### ISSUE TITLE
Remotes: Clustered Services

### ISSUE THREAD

**Issue by omidb** [getkyo/kyo#31]
I know it might not be at all relevant to KYO project but it can be great to consider the idea of running tasks cross machines from day one. In most Scala effect, ... projects running on large distributed system is always an after thought. A project that have a very nice designs for this is: [Ray](https://www.ray.io/)

**Comment by fwbrasil** [comment]
Agree! I think [aeron](https://github.com/real-logic/aeron) would be a solid foundation for that. It's designed for high-performance, availability, and provides a scalable pubsub.

**Comment by omidb** [comment]
some very interesting design ideas are being implemented here as well: https://restate.dev/

**Comment by hearnadam** [comment]
We have kyo-aeron now! Feel free to reopen if you have something more specific in mind. @omidb
